### ISSUE TITLE
'spring.profiles.active' JNDI option muzz lead to enabling specified profile, not to looking-up for 'spring.profiles.default' value

### ISSUE THREAD

**Issue by powerbroker** [spring-projects/spring-framework#33236]
**Affects:** 6.1.11

---

enable 'FINE' logging level of Spring.

1. Specify following in Tomcat context.xml

`<Environment name="spring.profiles.active" value="web" type="java.lang.String" override="false"/>`

start web-app with Spring(6.1.11) and enjoy 

`Looking up JNDI object with name [java:comp/env/spring.profiles.default] ...`

message series.

2. Add following to context.xml

`<Environment name="spring.profiles.default" value="web" type="java.lang.String" override="false"/>`

start web-app and enjoy 'Activating profiles [web]' in your log.

The 'spring.profiles.active' value muzz be used when set, ignoring the 'spring.profiles.default' one. The 'spring.profiles.default' value muzz take part when no 'spring.profiles.active' specified.

**Comment by snicoll** [comment]
> The 'spring.profiles.active' value muzz be used when set, ignoring the 'spring.profiles.default' one. The 'spring.profiles.default' value muzz take part when no 'spring.profiles.active' specified.

Yes, that's the expected behavior. In other words what you've stated is what should be happening. You're claiming it does not but XML snippet in text like that aren't really actionable as it doesn't make a working sample.

You could start by ignoring JNDI altogether and do the same thing in properties and validate that it works. If it does, then we're back on potentially a JNDI-specific problem and we'll need a small sample that demonstrates the problem we can run ourselves. You can attach a zip to this issue or push the code to a separate GitHub repository.
